libs/ktem/ktem/index/file/ui.py

```diff
@@ -296,12 +336,12 @@ def delete_event(self, file_id):
             for each in index:
                 if each[0].relation_type == "vector":
                     vs_ids.append(each[0].target_id)
-                else:
+                elif each[0].relation_type == "document":
                     ds_ids.append(each[0].target_id)
                 session.delete(each[0])
             session.commit()
 
-        if vs_ids:
+        if vs_ids and self._index._vs:
```

Comment:
hmm, haven't read the full impl of this file but access 2 private attributes at a time is smelly